PrimerSDK/Classes/User Interface/OAuth/OAuthViewModel.swift

```diff
@@ -88,34 +88,76 @@ class OAuthViewModel: OAuthViewModelProtocol {
                 switch result {
                 case .failure(let err): completion(err)
                 case .success(let res):
-                    instrument = PaymentInstrument(
```

Comment:
This function is huge with a lot of nested if-else statements. Let's split that up.

Reply:
busted! I'll add something a little more sophisticated 😇